Patch for kprop iprop full resync port contributed by Matt Rogers.

--- a/src/kadmin/server/ipropd_svc.c	
+++ a/src/kadmin/server/ipropd_svc.c	
@@ -255,6 +255,9 @@ ipropx_resync(uint32_t vers, struct svc_req *rqstp)
     gss_name_t name = NULL;
     char *client_name = NULL, *service_name = NULL;
     char *whoami = "iprop_full_resync_1";
+    int ipp = 0;
+    kadm5_config_params params;
+    char portstr[6];
 
     /*
      * vers contains the highest version number the client is
@@ -383,16 +386,25 @@ ipropx_resync(uint32_t vers, struct svc_req *rqstp)
 	    _exit(1);
 	}
 
-	DPRINT(("%s: exec `kprop -f %s %s' ...\n",
-		whoami, tmpf, clhost));
 	/* XXX Yuck!  */
 	if (getenv("KPROP_PORT"))
-	    pret = execl(KPROPD_DEFAULT_KPROP, "kprop", "-f", tmpf,
-			 "-P", getenv("KPROP_PORT"),
-			 clhost, NULL);
+		ipp = atoi(getenv("KPROP_PORT"));
 	else
+	    if (!kadm5_get_config_params(handle->context, 1, NULL, &params))
+	        ipp = params.iprop_port;
+
+	if (ipp != 0) {
+	    snprintf(portstr, sizeof(portstr), "%i", ipp);
+	    DPRINT(("%s: exec `kprop -f %s -P %s %s' ...\n", whoami, tmpf,
+				    portstr, clhost));
+	    pret = execl(KPROPD_DEFAULT_KPROP, "kprop", "-f", tmpf,
+			 "-P", portstr, clhost, NULL);
+	} else {
+	    DPRINT(("%s: exec `kprop -f %s %s' ...\n", whoami, tmpf, clhost));
 	    pret = execl(KPROPD_DEFAULT_KPROP, "kprop", "-f", tmpf,
 			 clhost, NULL);
+	}
+
 	if (pret == -1) {
 	    if (nofork) {
 		perror(whoami);
